src/generic/components/ecl-navigation/ecl-navigation-side/ecl-navigation-side.scss

```diff
@@ -0,0 +1,253 @@
+/**
+ * ECL Side menus
+ * @define navigation-side
+ */
+@mixin navigation-side() {
```

Comment:
I'm personally not a big fan of encapsulating too many rules or logic in a single block of code. This potentially limits the flexibility on "client implementation" side. If possible, the rules could be made in smaller pieces, so the "client implementation" can "compose" in smaller sizes.

Reply:
My way of thinking was that as long as the code is exactly the same on the generic and systems component, we could keep it as one block. As soon as there is a modification to be done we should decompose it to smaller mixins.

But this can be discussed.

Reply:
I agree with @kalinchernev that the rules could be made in smaller pieces 👍 But we have not defined clear conventions about that yet (I'll create a ticket for it), so let's not make this topic a blocker for this PR.